src/sentry/integrations/repositories.py

```diff
@@ -32,7 +31,7 @@ def check_file(self, repo, filepath, branch):
         try:
             self.get_client().check_file(repo, filepath, branch)
         except ApiError as e:
-            if e.code != 404:
+            if e.code not in [404, 400]:
```

Comment:
where are you getting 400s from?

Reply:
@MeredithAnya https://docs.gitlab.com/ee/api/repository_files.html#update-existing-file-in-repository:~:text=If%20the%20commit%20fails%20for%20any,file_path%20contained%20%2F..%2F%20(attempted%20directory%20traversal)%3B

Reply:
hmm its only documented for the PUT request but we got the same error here: https://sentry.io/organizations/sentry/issues/2176915130/?project=1&referrer=jira_integration

Reply:
@NisanthanNanthakumar @MeredithAnya I remember when I built the MVP, I would see `400`s from Gitlab for a file that didn't exist. But I think we should check the `400` case only for Gitlab (if Github gives us a `400`, that means something else is broken).

Reply:
Found a few more that might be interesting:
* https://sentry.io/organizations/sentry/issues/2206246887/?project=1&referrer=slack (a 403)
* https://sentry.io/organizations/sentry/issues/2206260381/?project=1&referrer=slack (a 500)

What do you think about ignoring those too?

Reply:
I think we should still re-raise for 403s, 500s, etc since they don't tell us if the file exists or not.